library_generation/utilities.sh

```diff
@@ -0,0 +1,110 @@
+#!/usr/bin/env bash
+
+set -xe
+
+# define utility functions
+extract_folder_name() {
+  destination_path=$1
+  folder_name=${destination_path##*/}
+  echo "$folder_name"
+}
+remove_empty_files() {
+  category=$1
+  find "$destination_path/$category-$folder_name/src/main/java" -type f -size 0 | while read -r f; do rm -f "${f}"; done
+  if [ -d "$destination_path/$category-$folder_name/src/main/java/samples" ]; then
+      mv "$destination_path/$category-$folder_name/src/main/java/samples" "$destination_path/$category-$folder_name"
+  fi
+}
+
+mv_src_files() {
```

Comment:
Do we have logic that moves the resource name helper classes from gapic folder to proto folder? Like this [file](https://github.com/googleapis/googleapis-gen/blob/master/google/bigtable/v2/google-cloud-bigtable-v2-java/proto-google-cloud-bigtable-v2-java/src/main/java/com/google/bigtable/v2/TableName.java)

Reply:
Interesting, do you know how this gapic-generate-java generated [file](https://github.com/googleapis/googleapis-gen/blob/master/google/bigtable/v2/google-cloud-bigtable-v2-java/proto-google-cloud-bigtable-v2-java/src/main/java/com/google/bigtable/v2/TableName.java) is being moved to the proto folder?

Reply:
After unzip `java_gapic_srcjar_raw.srcjar.zip` produced by the generator, [TableName.java](https://github.com/googleapis/googleapis-gen/blob/master/google/bigtable/v2/google-cloud-bigtable-v2-java/proto-google-cloud-bigtable-v2-java/src/main/java/com/google/bigtable/v2/TableName.java) is in `java_gapic_srcjar/proto/src/..`.

The command `mv_src_files "proto" "main"` is used to move this file to `proto-*/src/main/java/`.

I can see [InstanceName.java](https://github.com/googleapis/googleapis-gen/blob/master/google/bigtable/v2/google-cloud-bigtable-v2-java/proto-google-cloud-bigtable-v2-java/src/main/java/com/google/bigtable/v2/InstanceName.java) is also moved to `proto-*/src/main/java/` in the same way.

Reply:
Thanks! Can you make it clear by adding more comments regarding what files the scripts is moving and why?

Reply:
I added comments in `generate_library.sh` to explain which files are moved to which directory. PTAL.

About the reason, I copy these commands from [`java_gapic_pkg.bzl`](https://github.com/googleapis/sdk-platform-java/blob/main/rules_java_gapic/java_gapic_pkg.bzl#L252), I guess this is how GAPIC library is organized?